ci/ray_ci/tester_container.py

```diff
@@ -110,6 +110,7 @@ def run_tests(
     def _persist_test_results(self, team: str, bazel_log_dir: str) -> None:
         if os.environ.get("BUILDKITE_BRANCH") != "master":
             logger.info("Skip upload test results. We only upload on master branch.")
+            return
```

Comment:
could you follow up with some unit tests?

I think we have some bad patterns of reading env vars too much in the middle of the programs. similar to the argument that global vars are bad. ideally, they should be read at the beginning and being passed in.

Reply:
totally